### ISSUE TITLE
Output tensor of torch.split_with_sizes_copy has version 0 on CUDA while version 1 on CPU/XPU.

### ISSUE THREAD

**Issue by zhangxiaoli73** [pytorch/pytorch#136303]
### 🐛 Describe the bug

Hi, I found output tensor of `torch.split_with_sizes_copy` has version 0 on CUDA while has version 1 on CPU and XPU.

**Then what's the correct version of output tensor from `torch.split_with_sizes_copy`?**  version 0 or version 1?

I've put a minimal example to reproduce down below.

```python
import torch

world_size = 1
device_type="cuda"

all_gather_output = torch.randn((288), device=device_type)
all_gather_output = all_gather_output.view(world_size, -1)

out = [torch.randn((1, 256), device=device_type), torch.randn((1, 32), device=device_type)]
all_gather_input_split_sizes = [256, 32]

print(f"before version: {out[0]._version}")
torch.split_with_sizes_copy(all_gather_output, all_gather_input_split_sizes, dim=1, out=out)
print(f"after version: {out[0]._version}")
```
### Versions

Collecting environment information...
PyTorch version: 2.6.0.dev20240917+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.10.0 | packaged by conda-forge | (default, Nov 20 2021, 02:24:10) [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-97-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: 12.4.131
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA A100-PCIE-40GB
Nvidia driver version: 550.54.15
cuDNN version: Probably one of the following:
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.6.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      48 bits physical, 48 bits virtual
CPU(s):                             64
On-line CPU(s) list:                0-63
Thread(s) per core:                 1
Core(s) per socket:                 64
Socket(s):                          1
NUMA node(s):                       8
Vendor ID:                          AuthenticAMD
CPU family:                         25
Model:                              1
Model name:                         AMD EPYC 7713 64-Core Processor
Stepping:                           1
Frequency boost:                    enabled
CPU MHz:                            1500.000
CPU max MHz:                        3720.7029
CPU min MHz:                        1500.0000
BogoMIPS:                           4000.07
Virtualization:                     AMD-V
L1d cache:                          2 MiB
L1i cache:                          2 MiB
L2 cache:                           32 MiB
L3 cache:                           256 MiB
NUMA node0 CPU(s):                  0-7
NUMA node1 CPU(s):                  8-15
NUMA node2 CPU(s):                  16-23
NUMA node3 CPU(s):                  24-31
NUMA node4 CPU(s):                  32-39
NUMA node5 CPU(s):                  40-47
NUMA node6 CPU(s):                  48-55
NUMA node7 CPU(s):                  56-63
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Mitigation; safe RET
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 invpcid_single hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm

Versions of relevant libraries:
[pip3] numpy==2.1.0
[pip3] pytorch-triton==3.1.0+5fe38ffd73
[pip3] torch==2.6.0.dev20240917+cu124
[pip3] torchaudio==2.5.0.dev20240917+cu124
[pip3] torchvision==0.20.0.dev20240917+cu124
[pip3] triton==3.0.0
[conda] numpy                     2.1.0                    pypi_0    pypi
[conda] pytorch-triton            3.1.0+5fe38ffd73          pypi_0    pypi
[conda] torch                     2.6.0.dev20240917+cu124          pypi_0    pypi
[conda] torchaudio                2.5.0.dev20240917+cu124          pypi_0    pypi
[conda] torchvision               0.20.0.dev20240917+cu124          pypi_0    pypi
[conda] triton                    3.0.0                    pypi_0    pypi

cc @gujinghui @EikanWang @fengyuan14 @guangyey

**Comment by zhangxiaoli73** [comment]
@yifuwang It seems CUDA/CPU version difference comes from that you add an special implementation for CUDA. Then should  CUDA fast path also generate the same version as CPU?

**Comment by awgu** [comment]
https://github.com/pytorch/pytorch/issues/132014
I think no one has gotten around to fixing it.

**Comment by zhangxiaoli73** [comment]
@awgu Thank you to guide me to #132014. We found this issue when we try to enable FSDP2 on Intel XPU device.  

The version bump on XPU (same as CPU) will further lead to backward failures because of  version counter verification. So we are not asking for a version bump fix but how to handle those no-returns ops in autograd in PyTorch.

I saw some discussion in #132014 and will leave comments there.
